sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClientLiveTests.cs

```diff
@@ -207,10 +207,16 @@ public async Task AnomalyAlertConfigurationOperations()
 
             // Validate that we can Get the newly created config
             AnomalyAlertConfiguration getAlertConfig = await adminClient.GetAnomalyAlertConfigurationAsync(createdAlertConfig.Id).ConfigureAwait(false);
-            Response<IReadOnlyList<AnomalyAlertConfiguration>> getAlertConfigs = await adminClient.GetAnomalyAlertConfigurationsAsync(createdAnomalyDetectionConfiguration.Id).ConfigureAwait(false);
+
+            List<AnomalyAlertConfiguration> getAlertConfigs = new List<AnomalyAlertConfiguration>();
+
+            await foreach (var config in adminClient.GetAnomalyAlertConfigurationsAsync(createdAnomalyDetectionConfiguration.Id))
```

Comment:
Me learning, is this the only way to get the list of items after they are in the `AsyncPageable` type?

Reply:
It's the only "natural" approach, as far as I know. AsyncPageable implements the IAsyncEnumerable interface, and that's why it's accessed like this (more info about them [here](https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8)).

I think you could do something like this as well:
```cs
var enumerator = adminClient.GetSomethingAsync(...).GetAsyncEnumerator();

while (await enumerator.MoveNextAsync())
{
  var element = enumerator.Current;
  // Do something with <element>
}
```

That's probably what happens under the hood. I don't see a good reason for preferring this approach, though (maybe if you're using an older version of C# and can't use `await foreach`).

Reply:
I remembered I saw this somewhere once in [appconfig](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs#L824). You could use [ToEnumerableAsync](https://github.com/Azure/azure-sdk-for-net/blob/485e510817ec5729700ab29f984712f0f381ab84/sdk/core/Azure.Core.TestFramework/src/TestAsyncEnumerableExtensions.cs#L11).

Reply:
This is only a convenience method in our test framework, though. Good to know.

I thought you were asking from a user's point of view.